web/src/lib/api.ts

```diff
@@ -92,25 +88,23 @@ export default class API {
   /**
    * Ask the server for a clip
    */
-  getRandomClip(): Promise<Clip> {
-    return this.fetch('upload/random/', { responseType: 'blob', headers: {'uid': this.user.getId()}})
-      .then((req: XMLHttpRequest) => {
-        let src = window.URL.createObjectURL(req.response);
-        let glob = decodeURIComponent(req.getResponseHeader('glob'));
-        let sentence = decodeURIComponent(req.getResponseHeader('sentence'));
-        return Promise.resolve({ glob: glob, audio: src, sentence: sentence });
-      });
+  async getRandomClip(): Promise<Clip> {
+    const req = await this.fetch('upload/random/', { responseType: 'blob', headers: {'uid': this.user.getId()}})
```

Comment:
This was in response to my request, to use const unless necessary.

My reasoning is showing your intent - if you assume a variable to not be re-assigned, make it const, and only use let if you really need to re-assign. (I also prefer to use immutable structures where possible, so that philosophy goes along the same line.)
IOW, when touching existing code, const tells me I don't need to look around for any re-assignments of the identifier.

Reply:
Yes, I agree with @nevik about usage of `const` unless `let` is necessary.

One benefit as I see it is so that the static analyzer (in this case, TypeScript) will throw error if we reassign the variable somewhere we don't expect the variable to be reassigned.

Reply:
Alright, I'm convinced by both your arguments. Thanks for the info!